### PR TITLE
Cypress Automation Framework Refactoring

### PR DESCRIPTION
**Pull request by kingshuk-nandy**
The following features has been deployed

- Rearranged the Test case files of the cypress/integration folder as per their functionality into specific folders.
- Created a config folder that contains the different suite configurations
- Modified the Reports a bit. Now. screenshots are getting attached to the reports

### CODE REVIEW

File: cypress/integration/infrastructure/clusterUpgradation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/**
```
Comment (jruddell) [inline cypress/integration/infrastructure/clusterUpgradation.spec.ts:1]: why is this called upgradation? it should be called `cluster-upgrade.spec.ts`

File: cypress/integration/infrastructure/clusterUpgradation.spec.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * @testCaseId TC-INFRA-8
+ * @description Upgrading A Cluster
+ */
+
+import { loadConfig } from "cypress/support/commands"
+import { Config } from "cypress/support/types"
+
+describe("Cluster upgradation Flow", () => {
```
Comment (jruddell) [inline cypress/integration/infrastructure/clusterUpgradation.spec.ts:9]: upgradation -> upgrade

File: cypress/integration/infrastructure/multiMasterClusterCreation.spec.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
+/**
+ * @testCaseId TC-INFRA-1
+ * @description BareOS-Multi-Master-Cluster Creation
+ */
+
+import { loadConfig } from "cypress/support/commands"
+import { Config } from "cypress/support/types"
+
+describe("Multi Master Cluster Creation", () => {
```
Comment (jruddell) [inline cypress/integration/infrastructure/multiMasterClusterCreation.spec.ts:9]: stick with present tense for names please. `Multi-Master Cluster Create Flow`

File: cypress/integration/infrastructure/multiMasterClusterCreation.spec.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      .should("contain", "Select nodes to add as Worker Nodes")
+    cy.selectWorkerNodes(config.multiMasterClusterData.workerNodes)
+    cy.getByTestId("arrow-next").click()
+  })
+  it("should be able to setup Network", () => {
+    cy.getByTestId("cluster-api-fqdn").should("contain", "Cluster API FQDN")
+    cy.get("#masterVipIpv4").type(config.multiMasterClusterData.virtualIP)
+    cy.get("#masterVipIface").click()
+    cy.getByTestId("mastervipiface-value").last().click()
+    // TODO: Add validations
```
Comment (jruddell) [inline cypress/integration/infrastructure/multiMasterClusterCreation.spec.ts:60]: we should do the TODOs

File: cypress/integration/infrastructure/multiMasterClusterCreation.spec.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    cy.getByTestId("arrow-next").click()
+  })
+  it("should be able to setup Advanced Configuration", () => {
+    cy.getByTestId("advanced-configuration").should(
+      "contain",
+      "Advanced Configuration"
+    )
+    // TODO: Add validations
```
Comment (jruddell) [inline cypress/integration/infrastructure/multiMasterClusterCreation.spec.ts:68]: todo

File: cypress/integration/workloads/deployment.spec.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+    cy.wait(['@getClusters', '@getNamespaces'], { timeout: 60000 }).spread(
+      (getClusters, getNamespaces) => {
+        assert.equal(getClusters.response.statusCode, 200)
+        assert.equal(getNamespaces.response.statusCode, 200)
+      },
+    )
+    cy.getByTestId('deployments').click()
+
+    cy.intercept('**/clusters').as('getClusters')
```
Comment (jruddell) [inline cypress/integration/workloads/deployment.spec.ts:30]: why are we intercepting /clusters twice in this test?

File: package.json [hunk 7]
```diff
@@ -178,8 +178,8 @@
     "test:client:watch": "cross-env NODE_ENV=test jest --watch src/api-client",
     "test:e2e": "cross-env NODE_ENV=test jest -i  e2e-tests/",
     "test:integration": "cross-env NODE_ENV=test cypress run",
-    "cypress:open": "cross-env CYPRESS_ENV=test cypress open",
-    "cypress:run": "cross-env CYPRESS_ENV=test cypress run",
+    "cypress:open": "cross-env CYPRESS_ENV=test CYPRESS_baseUrl=http://localhost:3000/ cypress open --config-file cypress/config/smoke_suite.json",
```
Comment (jruddell) [inline package.json:181]: id prefer to leave the baseurl out of the command if possible. can't this be in the cypress.json file?

File: cypress/integration/workloads/pods.spec.ts [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    const clusterSelectors: SelectorDetails = {
+      id: 'clusterId',
+      dataTestId: 'clusterid-value',
+      itemToBeSelected: config.singleMasterClusterData.clusterName,
+      index: 0,
+    }
+    cy.selectFromDropdown(clusterSelectors)
+
+    cy.intercept('**/namespaces').as('getNamespaces')
+    cy.wait('@getNamespaces', { timeout: 40000 }).then((getDeployments) => {
```
Comment (knnguy) [inline cypress/integration/workloads/pods.spec.ts:53]: Should bet getNamespaces

File: cypress/integration/workloads/service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+  it('should be able to navigate to Services', () => {
+    cy.getByTestId('workloads', 9000).click()
+    cy.getByTestId('services').click()
+
+    cy.intercept('**/clusters').as('getClusters')
+    cy.intercept('**/services').as('getServices')
+    cy.intercept('**/namespaces').as('getNamespaces')
+
+    cy.wait(['@getServices', '@getClusters', '@getNamespaces'], { timeout: 60000 }).spread(
+      (getDeployments, getClusters, getNamespaces) => {
```
Comment (knnguy) [inline cypress/integration/workloads/service.spec.ts:25]: Should be getServices, not getDeployments

File: cypress/integration/infrastructure/clusterUpgradation.spec.ts (hunk 1 above)
Comment (kingshuk-nandy) [reply cypress/integration/infrastructure/clusterUpgradation.spec.ts]: renamed the file

File: cypress/integration/infrastructure/clusterUpgradation.spec.ts (hunk 2 above)
Comment (kingshuk-nandy) [reply cypress/integration/infrastructure/clusterUpgradation.spec.ts]: done

File: cypress/integration/infrastructure/multiMasterClusterCreation.spec.ts (hunk 3 above)
Comment (kingshuk-nandy) [reply cypress/integration/infrastructure/multiMasterClusterCreation.spec.ts]: Yes. renamed the describe.

File: cypress/integration/infrastructure/multiMasterClusterCreation.spec.ts (hunk 4 above)
Comment (kingshuk-nandy) [reply cypress/integration/infrastructure/multiMasterClusterCreation.spec.ts]: removed the TODOs

File: cypress/integration/infrastructure/multiMasterClusterCreation.spec.ts (hunk 5 above)
Comment (kingshuk-nandy) [reply cypress/integration/infrastructure/multiMasterClusterCreation.spec.ts]: done

File: cypress/integration/workloads/deployment.spec.ts (hunk 6 above)
Comment (kingshuk-nandy) [reply cypress/integration/workloads/deployment.spec.ts]: Previously it was required. But now with a lesser number of API calls it is not required. So we have removed it.

File: package.json (hunk 7 above)
Comment (kingshuk-nandy) [reply package.json]: Removed the baseUrl out of the command.

File: cypress/integration/workloads/deleteNamespace.spec.ts [hunk 10]
```diff
@@ -15,7 +15,14 @@ describe('Delete Namespace Flow', () => {
   it('should be able to delete a namespace', () => {
     cy.getByTestId('workloads').click()
     cy.getByTestId('namespaces').click()
+
+    cy.intercept('**/namespaces').as('getNamespaces')
+
+    cy.wait('@getNamespaces', { timeout: 40000 }).then((getDeployments) => {
```
[the remark this answers is not in the document]
Comment (kingshuk-nandy) [reply cypress/integration/workloads/deleteNamespace.spec.ts]: renamed the same

File: cypress/integration/workloads/pods.spec.ts (hunk 8 above)
Comment (kingshuk-nandy) [reply cypress/integration/workloads/pods.spec.ts]: done

File: cypress/integration/workloads/service.spec.ts (hunk 9 above)
Comment (kingshuk-nandy) [reply cypress/integration/workloads/service.spec.ts]: done

File: cypress/integration/workloads/service.spec.ts [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+    const clusterSelectors: SelectorDetails = {
+      id: 'clusterId',
+      dataTestId: 'clusterid-value',
+      itemToBeSelected: config.singleMasterClusterData.clusterName,
+      index: 0,
+    }
+    cy.selectFromDropdown(clusterSelectors)
+
+    cy.intercept('**/namespaces').as('getNamespaces')
+    cy.wait('@getNamespaces', { timeout: 40000 }).then((getDeployments) => {
```
[the remark this answers is not in the document]
Comment (kingshuk-nandy) [reply cypress/integration/workloads/service.spec.ts]: done

**Review by jruddell — APPROVED** [review]

**Pull request merged**